eclass/flag-o-matic.eclass

```diff
@@ -372,19 +380,30 @@ replace-cpu-flags() {
 # @USAGE: <variable> <flag>
 # @INTERNAL
 # @DESCRIPTION:
-# Returns shell true if <flag> is in a given <variable>, else returns shell false.
+# Returns shell true if <flag> is in a given <variable> and not surpassed by e.g. -fno-<flag>,
+# else returns shell false.
 _is_flagq() {
-	local x var="$1[*]"
+	local x var="$1[*]" equal_str equal_pos flag_no active=1
+	equal_str=${2%%=*}
+	equal_pos=${#equal_str}
+	# This results in -fno-flag for -fflag=*
+	flag_no="${2:0:2}no-${2:2:(equal_pos - 2)}"
```

Comment:
fwiw quotes, spaces, and parentheses are unneeded, could even drop the 0 (2::2). But it's all style and debatable so not necessary to change all (the parentheses does feel a bit out of place though). Pos stuff could use `${equal_str:2}` too (just removes -f on the already-stripped of =* string).

That aside, I think should work on `${x}` rather than `$2` because of things like (from chromium in this case):
`if ( shopt -s extglob; is-flagq '-g?(gdb)?([1-9])' ); then`

Reply:
May work fine sometimes, but trying to modify (ext)glob is likely to miss things.

Reply:
And right, if use ${x} will need to check if it matched first too.

Reply:
Hmm.. actually having second thoughts, maybe(?) $2 works fine as a best effort.

Technically this can be used to check more than one flag at once (any of), and I guess no- will get weird either way. Does make me wonder if doing this is a good idea.

Reply:
I'm okay with best effort, leaning towards more readable Bash. If someone insists on -fno -f -fno ... then I think it's not unreasonable that something might trip up.

Reply:
> That aside, I think should work on `${x}` rather than `$2` because of things like (from chromium in this case): `if ( shopt -s extglob; is-flagq '-g?(gdb)?([1-9])' ); then`

Wouldn't that still work? Besides, chromium and rust seem to be the only users of extglob + is-flagq.

Of course, all of this is best-effort. E.g. this will still misdetect `CFLAGS="-O3 -O2" is-flagq -O3"

> I'm okay with best effort, leaning towards more readable Bash. If someone insists on -fno -f -fno ... then I think it's not unreasonable that something might trip up.

But this is rather common. For example it's the approach the wiki recommends for many things such as LTO.

Reply:
I think I'm fine with trying because if someone is doing stupid things, they're going to find their way around anyway.